packages/react/src/lib/renderComponent.tsx

```diff
@@ -248,22 +242,40 @@ const renderComponent = <P extends {}>(
     theme,
   }
 
-  if (accessibility.focusZone) {
-    return renderWithFocusZone(render, accessibility.focusZone, resolvedConfig)
-  }
-
   // conditionally add sources for evaluating debug information to component
   if (isDebugEnabled) {
-    saveDebug(
-      new Debug({
-        componentName: displayName,
-        themes: context ? context.originalThemes : [],
-        instanceStylesOverrides: props.styles,
-        instanceVariablesOverrides: props.variables,
-        resolveStyles: styles => resolveStyles(styles, styleParam),
-        resolveVariables: variables => callable(variables)(theme.siteVariables),
+    saveDebug({
+      componentName: displayName,
+      componentVariables: _.filter(
+        resolvedVariables._debug,
+        variables => !_.isEmpty(variables.resolved),
+      ),
+      componentStyles: _.mapValues(resolvedStylesDebug, v =>
+        _.filter(v, v => {
+          return !_.isEmpty(v.styles)
+        }),
+      ),
+      siteVariables: _.filter(theme.siteVariables._debug, siteVars => {
+        if (_.isEmpty(siteVars) || _.isEmpty(siteVars.resolved)) {
+          return false
+        }
+
+        const keys = Object.keys(siteVars.resolved)
+        if (
+          keys.length === 1 &&
+          keys.pop() === 'fontSizes' &&
+          _.isEmpty(siteVars.resolved['fontSizes'])
+        ) {
+          return false
+        }
+
+        return true
       }),
-    )
+    })
```

Comment:
Should be moved out somehow? Can we keep this logic in `new Debug()`, `filterDebug()`, `formatDebugData()`, etc?